src/base/memory_cwrapper.h

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_cwrapper.h - managing memory with allocating wrapper functions
+ *                     and memory monitor
+ */
+
+#ifndef _MEMORY_CWRAPPER_H_
+#define _MEMORY_CWRAPPER_H_
+
+#ifdef SERVER_MODE
+#include <string.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <assert.h>
+
+#include "memory_monitor_sr.hpp"
+
+inline size_t
+get_allocated_size (void *ptr)
+{
+  if (ptr == NULL)
+    {
+      return 0;
+    }
+  else
+    {
+      return mmon_get_allocated_size ((char *) ptr);
+    }
+}
+
+inline void
+cub_free (void *ptr)
+{
+  if (mmon_is_memory_monitor_enabled () && ptr != NULL)
+    {
+      mmon_sub_stat ((char *) ptr);
+      assert (malloc_usable_size (ptr) != 0);
```

Comment:
It seems to be a precondition, not postcondition. Then, it should be above `mmon_sub_stat`. Furthermore, I think it should be in the `mmon_sub_stat` if it's the case.